services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/UpdateHakemusITest.kt

```diff
@@ -506,26 +502,55 @@ class UpdateHakemusITest(
 
         @Test
         fun `throws exception when area is not inside hanke area`() {
-            val hanke = hankeFactory.builder(USERNAME).withHankealue().saveEntity()
+            val hanke = hankeFactory.builder(USERNAME).withHankealue().save()
+            val hankeEntity = hankeRepository.findAll().single()
             val hakemus =
                 hakemusFactory
-                    .builder(USERNAME, hanke, ApplicationType.EXCAVATION_NOTIFICATION)
+                    .builder(USERNAME, hankeEntity, ApplicationType.EXCAVATION_NOTIFICATION)
                     .save()
-            val request = hakemus.toUpdateRequest().withAreas(listOf(notInHankeArea))
+            val hankealueId = hanke.alueet.single().id!!
+            val area = notInHankeArea.copy(hankealueId = hankealueId)
+            val request = hakemus.toUpdateRequest().withArea(area)
 
             val exception = assertFailure {
                 hakemusService.updateHakemus(hakemus.id, request, USERNAME)
             }
 
             exception.all {
                 hasClass(HakemusGeometryNotInsideHankeException::class)
-                messageContains("id=${hakemus.id}")
-                messageContains(hanke.logString())
+                messageContains("Hakemus geometry is outside the associated hankealue")
+                messageContains("hankealue=$hankealueId")
                 messageContains(
                     "geometry=${notInHankeArea.tyoalueet.single().geometry.toJsonString()}")
             }
         }
 
+        @Test
+        fun `throws exception when area references non-existent hankealue`() {
+            val hanke = hankeFactory.builder(USERNAME).withHankealue().save()
+            val hankeEntity = hankeRepository.findAll().single()
+            val hakemus =
+                hakemusFactory
+                    .builder(USERNAME, hankeEntity, ApplicationType.EXCAVATION_NOTIFICATION)
+                    .withWorkDescription("Old work description")
```

Comment:
Is work description needed in this test?

Reply:
It's not. The hakija isn't either. I removed them.